### ISSUE TITLE
ReST lexer provides inconsistent results

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#635]
(Original issue 928 created by chebee7i on 2013-11-16T18:57:30.727979+00:00)

Although 'code-block' and 'sourcecode' are aliases to the `CodeBlock` directive, there seems to be an issue with how nested blocks are parsed.  In the following snippet, 

```
#!rst

The code in the `sourcecode` directive is tagged `Token.Literal.String`:

.. code-block:: rst

    .. sourcecode:: ipythoncon

        In [1]: a = 'foo'
        In [2]: b = 1


The code in the inner `code-block` is tagged `Token.Punctuation`:

.. code-block:: rst

    .. code-block:: ipythoncon

        In [1]: a = 'foo'
        In [2]: b = 1

So it seems that the first block is correct.  
```

**Comment by Anteru** [comment]
(Original issue 928 last updated on 2014-01-09T18:08:27.758594+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by chebee7i on 2013-11-16T18:58:40.261907+00:00)

Or maybe they are both incorrect?

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-01-09T18:08:27.749692+00:00)

Closes #928: now also code-block directives are handled specially.

→ <<cset 5dd367b461c7>>
